### PR TITLE
📖 Refactor core developer guide

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:

Refactor core developer guide by bringing everything under the same section

Please note that this PR is just shuffling pages around, and a couple of follow up PRs are required to get to to a cleaner separation between core development and provider development content

/area documentation

### CODE REVIEW

File: docs/book/src/developer/core/overview.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Developing "core" Cluster API
+
+This section of the book is about developing "core" Cluster API.
+
+Whith "core" Cluster API we refer to the common set of API and controllers that are required to run
```
[the remark this answers is not in the document]
Comment (fabriziopandini) [reply docs/book/src/developer/core/overview.md]: fixed

File: docs/book/src/reference/glossary.md [hunk 2]
```diff
@@ -174,13 +177,27 @@ Control plane provider's interaction with Cluster API is based on what is define
 
 See [KCP](#kcp).
 
+### Core Cluster API
+
+Whith "core" Cluster API we refer to the common set of API and controllers that are required to run
```
[the remark this answers is not in the document]
Comment (fabriziopandini) [reply docs/book/src/reference/glossary.md]: fixed

**Review by chrischdi — COMMENTED** [review]
/lgtm

**Pull request merged**